scripts/barclamp_mgmt_lib.rb

```diff
@@ -408,89 +390,47 @@ def bc_install_layout_1_app(from_rpm, bc, bc_path, yaml)
 
     if dirs.include? 'crowbar_framework'
       debug "path entries include \"crowbar_framework\""
-      files += bc_cloner('crowbar_framework', bc, nil, bc_path, BASE_PATH, false)
-      framework_permissions bc, bc_path
+      files += bc_cloner('crowbar_framework', nil, bc_path, BASE_PATH)
+      framework_permissions
     end
 
     if dirs.include? 'bin'
       debug "path entries include \"bin\""
-      files += bc_cloner('bin', bc, nil, bc_path, BASE_PATH, false)
+      files += bc_cloner('bin', nil, bc_path, BASE_PATH)
       FileUtils.chmod_R 0755, BIN_PATH
       debug "\tcopied command line files"
     end
 
     if dirs.include? 'chef'
       debug "path entries include \"chef\""
-      files += bc_cloner('chef', bc, nil, bc_path, BASE_PATH, false)
+      files += bc_cloner('chef', nil, bc_path, BASE_PATH)
       debug "\tcopied over chef parts from #{bc_path} to #{BASE_PATH}"
     end
+
+    # copy over the crowbar YAML files, needed to update catalog
+    yml_path = File.join CROWBAR_PATH, 'barclamps'
+    get_yml_paths(bc_path).each do |yml_source|
+      yml_created = File.join(yml_path, File.basename(yml_source))
+      FileUtils.mkdir yml_path unless File.directory? yml_path
+      FileUtils.cp yml_source, yml_created unless yml_source == yml_created
+      files << yml_created
+    end
   end
 
   # we don't install these files in the right place from rpm
   if dirs.include? 'updates'
     debug "path entries include \"updates\""
-    files += bc_cloner('updates', bc, nil, bc_path, ROOT_PATH, false)
+    files += bc_cloner('updates', nil, bc_path, ROOT_PATH)
```

Comment:
Style/StringLiterals: Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping. (https://github.com/bbatsov/ruby-style-guide#consistent-string-literals)